### DISCUSSION TITLE
Guys you must really join your efforts with this guy maintaining the project  vidkidz.github.io

### DISCUSSION THREAD

**Discussion by ROBERT-MCDOWELL** [ruffle-rs/ruffle#973] [General]
his repo
https://github.com/vidkidz/vidkidz.github.io

his demo
vidkidz.github.io

I tested 10 games in AS2 and the result is absolutey insane, I really don't know how he did the magic but
99% of the tested games work flawessly, only one graphic bug on one game (snowboarding) when the character disappears when the race starts. Frankly this guy is talented.

**Comment by manokara** [comment]
From what I've seen, maybe this could be based on Crossbridge/Alchemy, so it's not really that magical :P But unfortunately it's not open source (yet?), so I'm not sure if a collaboration could happen. The guy does have a fork of ruffle though, hmm.

**Comment by ROBERT-MCDOWELL** [comment]
yeah it seems that he made a trick from crossbridge, which is not too bad IMHO.
I tested 10 games in As2 and frankly, it's impressive...

**Comment by ROBERT-MCDOWELL** [comment]
his demo page
https://vidkidz.github.io/
the crazy thing is it works also on windows xp with old version of mozilla or chrome!

**Comment by QualityLow** [comment]
The first random thing I tried, "Bart Simpson Island Escape.swf", failed and crashed on the loading screen with this message: "Ooops! An error occurred. Reload the page and try again."

Ruffle on the other hand is able to get past the loading screen. However the game is far from perfect when using Ruffle either. I think I've stumbled over a good AS2 game to try and get working with Ruffle. Some issues I see:

1. The white glow around Bart is missing on the title screen and loading screen.
2. Mute button in the top-right corner doesn't work.
3. The speech bubbles doesn't show up in the intro cutscene.
4. "A parachute" description not showing up when hovering it.
5. Bart gets a face mask after picking up the parachute, which shouldn't happen.
6. "A red button" description not showing up when hovering it.
7. Text positioning is wrong when you hover inventory items.
8. Clicking and dropping the chocolate on the tree doesn't make the boar go and eat it, which should happen.
9. Clicking Bart in the tree makes his speech bubble show up in the wrong place with halved text field width and/or freezes the browser.
10. General low performance (probably irrelevant since that's how it is with emulation).

If all of these are fixed (maybe skipping 1 and 10 for now) it would reflect well on many other AS2 games. Issue 5 and 8 are especially worrisome since there must be some kind of logic flaw in the AS2 interpreter.

I've tested using "ruffle_nightly_2020_11_25_selfhosted.zip". The flash in question can be downloaded from: http://eye.swfchan.com/flash.asp?id=138482&n=Bart+Simpson+Island+Escape.swf (SFW flash but there are NSFW ads on the site.)

**Comment by QualityLow** [comment]
His flash player is still pretty impressive (though the top-right watermark is annoying). I hope he plans on making it available to all and not only sell it or something.

I tried http://eye.swfchan.com/flash.asp?id=856&n=n.swf (AS2) and it works better in his player compared to Ruffle.
I tried http://eye.swfchan.com/flash.asp?id=31560&n=Pandemic2.swf (AS2) and it works better in his player compared to Ruffle (where it won't even load). His player works but doesn't have text glow and you can't name the virus.
I tried http://eye.swfchan.com/flash.asp?id=48108&n=Storm+the+House+3.swf (AS2) and it works in both, just some font and transparency issues (same in both).

His player seems to handle http://eye.swfchan.com/flash.asp?id=145374&n=Luftrauser.swf (AS3) pretty much perfectly.
His player seems to handle http://eye.swfchan.com/flash.asp?id=167131&n=the_best_amendment.swf (AS3) but not perfectly (lingering muzzle flashes).
His player seems to handle http://eye.swfchan.com/flash.asp?id=100924&n=esc.swf (AS3) but not perfectly (some color issues).
His player couldn't handle http://eye.swfchan.com/flash.asp?id=114647&n=Experiment+17.swf (AS3), crashes with the "Ooops" error.
His player couldn't handle http://eye.swfchan.com/flash.asp?id=108398&n=Death+Worm.swf (AS3), doesn't load (no error message).
His player also couldn't handle http://eye.swfchan.com/flash.asp?id=200035&n=Quake.swf (AS3), which isn't a surprise.

If there are no legal hurdles to do things the same way he has done things Ruffle could mimic how he has implemented AS3 (because hey if it works it works). Anyone tried contacting him? Maybe he would gladly share his code or at least his methods if only asked. He is on twitter: https://twitter.com/vidkidz/

**Comment by ROBERT-MCDOWELL** [comment]
yeah vidkidz made also a huge job but unfortunately no source code for now. Anyhow I wish his project to succeed like ruffle.

**Comment by ousia** [comment]
@ROBERT-MCDOWELL, it is a pity that @vidkidz doesn’t provide source code for _WAFlash_.

Cooperation among projects might benefit all. But it is up to him if he decides to publish the source code.

**Comment by ROBERT-MCDOWELL** [comment]
I don't think it's a pity, it's his choice, maybe he has some constraints, or he feels more comfortable to work as this. But I'm pretty sure that once he'll covered the whole as2/3 API he will release them.

**Comment by ousia** [comment]
If WAFlash could have its source code released under any free license, that would be great news.

Future will tell...

**Comment by HeckerCatYT** [comment]
I have tried A Game called zombie fight club from y8 and it dosent work. most games do not work at all. Ruffle is alot better at supporting Games then WaFlash.

**Comment by ROBERT-MCDOWELL** [comment]
today yes, unfortunately waflash author stopped his project one year ago.  no news since.

**Reply by n0samu** [reply]
WAFlash is still under development, the latest release was on 2023-08-25. The author does not allow it to be used on other sites besides his own [WAFlash Archive](https://vidkidz.tistory.com/). He doesn't announce releases of the actual emulator, only really focusing on game updates or fixes in the archive. You can follow news on that through the WAFlash Discord or [this page on the site](https://vidkidz.tistory.com/notice/1788). It's unfortunate how self-limited the WAFlash project is, with everything being only in Korean and exclusive to one small site.

**Reply by n0samu** [reply]
I just found that you can see a list of the newly added games on Tistory's mobile theme: https://vidkidz.tistory.com/m/. This shows that several games per day are being added to the site. And recently a save manager was also added, allowing people to import and export their save files for some of the site's popular games: https://vidkidz.tistory.com/pages/waflash-save-manager

**Reply by ROBERT-MCDOWELL** [reply]
mmhmm , ok so he preferred to work in secret so :)
would be interesting to se his code...

**Reply by ROBERT-MCDOWELL** [reply]
btw maybe he uses a headless chrome with a version using the last pepperflash and made a webassembly app to stream the game to the client?

**Reply by n0samu** [reply]
No, WAFlash is an emulator, pretty similar to Lightspark, but compiled to WebAssembly.

**Reply by ROBERT-MCDOWELL** [reply]
hey made a pretty good job anyhow. I wonder if they cover the network classes too
